app/queries/projects_query.rb

```diff
@@ -0,0 +1,29 @@
+class ProjectsQuery
+  attr_reader :relation
+
+  def initialize(relation = Project.all)
+    @relation = relation
+  end
+
+  def search_name_description(query)
+    if  ActiveRecord::Base.connection_config["adapter"] === "postgresql"
+      full_text_search_name_description(query)
+    else
+      simple_search_name_description(query)
+    end
+  end
+
+  def full_text_search_name_description(query)
+    relation.includes(:author).search(query).select("id,author_id,image_preview,name,description,view")
+  end
+
+  def simple_search_name_description(query)
+    relation.includes(:author).where("name LIKE ? OR description LIKE ?","%#{query}%","%#{query}%").select("id,author_id,image_preview,name,description,view")
```

Comment:
```ruby
  relation
    .includes(:author)
    .where("name LIKE ? OR description LIKE ?","%#{query}%","%#{query}%")
    .select(:id ,:author_id, :image_preview, :name, :description ,:view)
```

And you can make `where` clause more readable as

```ruby
  .where("name LIKE :query OR description LIKE :query", query: "%#{query}%")
```

Reply:
Yeah it is much more readable.Fixed:P

Reply:
@garry329 Not just the query syntax. Also make it multiline to make it more readable